ouroboros-consensus/src/Ouroboros/Storage/ChainDB/API.hs

```diff
@@ -582,7 +583,32 @@ data ChainDbFailure =
 
 deriving instance Show ChainDbFailure
 
-instance Exception ChainDbFailure
+instance Exception ChainDbFailure where
+  displayException = \case
+      ImmDbParseFailure {}                -> corruption
+      ImmDbTrailingData {}                -> corruption
+      ImmDbMissingBlock {}                -> corruption
+      ImmDbMissingBlockPoint {}           -> corruption
+      ImmDbUnexpectedIteratorExhausted {} -> corruption
+      ImmDbFailure e -> case e of
+        ImmDB.FileSystemError fse      -> fsError fse
+        ImmDB.InvalidFileError {}      -> corruption
+        ImmDB.MissingFileError {}      -> corruption
+        ImmDB.ChecksumMismatchError {} -> corruption
+      VolDbParseFailure {}                -> corruption
+      VolDbTrailingData {}                -> corruption
+      VolDbMissingBlock {}                -> corruption
+      VolDbFailure e -> case e of
+        VolDB.FileSystemError fse -> fsError fse
+        VolDB.ParserError {}      -> corruption
+      LgrDbFailure fse                    -> fsError fse
+      ChainDbMissingBlock {}              -> corruption
+    where
+      corruption = "The database got corrupted, please restart with validation mode enabled"
```

Comment:
I could mention the `--validate-db` flag I'm introducing in https://github.com/input-output-hk/cardano-node/pull/353, but I'm afraid it will get out of date/sync, as the flag is part of another repository.

Reply:
I agree, it would.